File: Array/secondLargest.py
def secondLargest(arr, n):
    if n < 2: 
        return -1
    large = arr[0]
    # large = float('-inf')
    second_large = float('-inf')
    for i in range(n):
        if arr[i] > large:
            second_large = large
            large = arr[i]
        elif arr[i] > second_large and arr[i] != large:
            second_large = arr[i]
    return second_large

# Driver code
if __name__ == "__main__":
    arr = [1, 2, 4, 7, 7, 5]  # Array of elements
    n = len(arr)  # Size of the array

    # Find the second smallest and second largest elements
    sL = secondLargest(arr, n)

    # Output the results
    print(f"Second largest is {sL}")

File: Array/CPP/secondLargest.cpp
#include<bits/stdc++.h>
using namespace std;


// #brute method
int secondLargest(vector<int>& arr){
    sort(arr.begin(), arr.end());
    int largest = arr[arr.size()-1];
    int secondLarge;
    for(int i = arr.size() - 1; i>= 0; i--){
        if(arr[i] != largest){
            secondLarge = arr[i];
            break; 
        }
    }
}

// # optimal Solution

int secondLargestO(vector<int>& arr){
    int largest = arr[0];
    for (int i = 0; i<= arr.size()-1; i++){
        if (arr[i] > largest && arr[i] != largest){
            largest = arr[i];
            // cout << largest;
        }
    }
    return largest;
}

int main(){
    vector<int> arr1 = {1, 4, 2, 5, 2};
    cout << secondLargestO(arr1);
    return 0;
}